rockpi4.mk

```diff
@@ -0,0 +1,239 @@
+COMPILE_NS_USER ?= 64
+override COMPILE_NS_KERNEL := 64
+COMPILE_S_USER ?= 64
+COMPILE_S_KERNEL ?= 64
+
+CFG_TEE_CORE_LOG_LEVEL ?= 2
+
+include common.mk
+
+DEBUG ?= 1
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+TF_A_PATH		?= $(ROOT)/trusted-firmware-a
+BINARIES_PATH		?= $(ROOT)/out
+UBOOT_PATH		?= $(ROOT)/u-boot
+UBOOT_BIN		?= $(UBOOT_PATH)/u-boot.bin
+ROOT_IMG 		?= $(ROOT)/out-br/images/rootfs.ext2
+BOOT_IMG		?= $(ROOT)/out/rockpi4.img
+RKDEVELOPTOOL_PATH	?= $(ROOT)/rkdeveloptool
+RKDEVELOPTOOL_BIN	?= $(RKDEVELOPTOOL_PATH)/rkdeveloptool
+LOADER_BIN		?= $(BINARIES_PATH)/rk3399_loader_v1.20.119.bin
+
+BR2_TARGET_ROOTFS_CPIO=n
+BR2_TARGET_ROOTFS_CPIO_GZIP=n
+BR2_TARGET_ROOTFS_EXT2=y
+BR2_TARGET_ROOTFS_EXT2_SIZE=112M
+BR2_TARGET_GENERIC_GETTY_PORT=ttyS2
+
+################################################################################
+# Targets
+################################################################################
+
+all: boot-img
+
+clean: buildroot-clean
+
+include toolchain.mk
+
+################################################################################
+# Arm Trusted Firmware-A
+################################################################################
+TF_A_EXPORTS ?= CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
+		M0_CROSS_COMPILE="$(CCACHE)$(AARCH32_CROSS_COMPILE)"
+
+TF_A_DEBUG ?= $(DEBUG)
+ifeq ($(TF_A_DEBUG),0)
+TF_A_LOGLVL ?= 30
+TF_A_OUT = $(TF_A_PATH)/build/rk3399/release
+else
+TF_A_LOGLVL ?= 40
+TF_A_OUT = $(TF_A_PATH)/build/rk3399/debug
+endif
+
+TF_A_FLAGS ?= ARCH=aarch64 PLAT=rk3399 SPD=opteed DEBUG=$(TF_A_DEBUG) \
+	      LOG_LEVEL=$(TF_A_LOGLVL)
+
+$(TF_A_PATH)/.stamp_patched:
+	patch -d $(TF_A_PATH) -p1 <$(ROOT)/build/rockpi4/tfa/0001-rk3399-set-console-baudrate-to-1500000.patch
```

Comment:
It this something in the works? I.e., looks like this is something that should be sent upstream, or?

Reply:
Hmm, someone has been "tainted" by yocto? ;-)

This must be quite annoying when working on TF-A.

Reply:
> Hmm, someone has been "tainted" by yocto? ;-)

Nooooo! I swear! :wink: 

> It this something in the works? I.e., looks like this is something that should be sent upstream, or?

Absolutely, I plan on sending a patch, but it may take some time until this gets into a release. Especially since I am not familiar with TF-A upstreaming at all (Gerrit etc.). In the meantime I thought this was an OK-ish solution -- but note that even without this patch things should work, just without the initial TF-A and OP-TEE logs.